### PR TITLE
Enable job e2e tests for GKE

### PR DESCRIPTION
**Pull request by j3ffml**
will need cherrypick onto 1.1

### CODE REVIEW

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -565,6 +571,7 @@ case ${JOB_NAME} in
     : ${FAIL_ON_GCP_RESOURCE_LEAK:="true"}
     : ${GINKGO_TEST_ARGS:="--ginkgo.skip=$(join_regex_allow_empty \
           ${GKE_REQUIRED_SKIP_TESTS[@]:+${GKE_REQUIRED_SKIP_TESTS[@]}} \
+          "Daemon\sset|Deployment" \
```
Comment (ihmccreery) [inline hack/jenkins/e2e.sh]: This is surprising to see; can you add a note here explaining why it is as it is?

**Comment by ihmccreery** [comment]
Thanks for doing this, @jlowdermilk!  One small request for comment; otherwise LGTM.

File: hack/jenkins/e2e.sh (hunk 1 above)
Comment (jlowdermilk) [inline hack/jenkins/e2e.sh]: Added comment. Daemon sets and Deployments are experimental (not beta) features in the 1.1. release, and turned off for GKE.

**Comment by jlowdermilk** [comment]
@k8s-bot test this

**Comment by jlowdermilk** [comment]
@k8s-bot test this
@k8s-bot unit test this

**Comment by jlowdermilk** [comment]
@k8s-bot test this

**Pull request merged**